emcc.py

```diff
@@ -974,12 +974,20 @@ def optimizing(opts):
 
     newargs = [a for a in newargs if a is not '']
 
-    # -c means do not link in gcc, and for us, the parallel is to not go all the way to JS, but stop at bitcode
     has_dash_c = '-c' in newargs
+    has_dash_S = '-S' in newargs
     if has_dash_c:
       assert has_source_inputs or has_header_inputs, 'Must have source code or header inputs to use -c'
       target = target_basename + '.o'
       final_suffix = '.o'
+    elif has_dash_S:
+      if '-emit-llvm' in newargs:
```

Comment:
I'm not sure it's worth checking for `-emit-llvm` - if we do that, we may also want to check `-flto` and `WASM_OBJECT_FILES=0` as those also affect whether we emit bitcode. But I think it might be overkill to do all that.

Reply:
This is all for the default output filename if otherwise unspecified by the user, right? So letting this be best-effort is probably good enough.

Reply:
Regardless of WASM_OBJECT_FILES or flto settings -emit-llvm will always generate bitcode output from clang, and this is already supported in way since we pass this through from the command line.

Or are you saying the `-emit-llvm` might not have come from the user? .. I'll check that.